examples/zombies.jl

```diff
@@ -0,0 +1,117 @@
+# # Zombie Outbreak
+# ![](outbreak.gif)
+#
+# This model showcases an ABM running on a map, using [`OpenStreetMapSpace`](@ref).
+#
+# ## Constructing the end of days
+using Agents
+using Random # hide
+
+# We'll simulate a zombie outbreak in a city. To do so, we start with an agent which
+# satisfies the OSMSpace conditions of having a position of type [`OSMPos`](@ref) and
+# a `route` vector.
+
+mutable struct Zombie <: AbstractAgent
+    id::Int # Required
+    pos::OSMPos # Required
+    route::Vector{Int} # Required
+    infected::Bool # User added
+end
+
+# The model constructor we build consists of a map, and 100 agents scattered randomly
+# around it. They have their own agenda and need to travel to some new destination.
+# Unfortunately one of the population has turned and will begin infecting anyone who
+# comes close.
+
+function initialise(; map_path = TEST_MAP)
+    model = ABM(Zombie, OpenStreetMapSpace(map_path))
+
+    for _ in 1:100
+        start = osm_random_direction(model) # Somewhere on a road
+        finish = random_position(model) # At an intersection
+
+        ## We already have our start and finish edges, so we identify the in-between
+        path = osm_plan_route(start[2], finish[1], model)
```

Comment:
I think we can streamline this to a single function `add_random_route!`. In the end that's what we care about, that each agent has a random route to follow.

And specifically we can make it `add_random_route!(start = random)`.

Reply:
Maybe true, but it'll just be a wrapper. `osm_plan_route` is far more important since I suspect that users will want to look at traffic flows and other data-driven solutions with this map space. In addition, there's a route planner *by distance* and one *by quickest* or something. We should allow access to both.

Reply:
of course `osm_plan_route` will be epxported, my suggestion is just for wrapping these 3 lines. Does the route planning already have a "quickest" and "by distance" options? And what does "by distance" even mean by the way?

Reply:
Fastest & shortest is actually the options. I've added the option for both now.